x-pack/filebeat/input/cel/input.go

```diff
@@ -758,6 +760,25 @@ func newClient(ctx context.Context, cfg config, log *logp.Logger, reg *monitorin
 		maxSize := cfg.Resource.Tracer.MaxSize * 1e6
 		trace = httplog.NewLoggingRoundTripper(c.Transport, traceLogger, max(0, maxSize-margin), log)
 		c.Transport = trace
+	} else if cfg.Resource.Tracer != nil {
```

Comment:
Looking at the code it seems like it's been the case that any `resource.tracer.*` config, say `resource.tracer.maxsize`, would be enough to activate the tracing, even without setting a filename.

Lumberjack can handle no filename. It "uses <processname>-lumberjack.log in os.TempDir() if empty." But we wouldn't have access to that filename in the config struct.

Now with `resource.tracer.enabled: false`, but no file name, we'd try call `os.Remove("")` and get an error.

I think the `cel.tracerConfig.enabled` function should require a file name. The clean-up case should check for a file name as well, either directly or maybe with something like `cel.tracerConfig.disabledWithFilename`.

Reply:
> Looking at the code it seems like it's been the case that any `resource.tracer.*` config, say `resource.tracer.maxsize`, would be enough to activate the tracing, even without setting a filename.

This would be true if the validation were not run; it [checks that the filename is present and fails if not](https://github.com/efd6/beats/blob/8952f2ca59b490e49461f2b288d7e04271fb8c73/x-pack/filebeat/input/cel/config.go#L254-L268).